File: introCS/python/sample.py
import sys
import random
import stdarray
import stdio

m = int(sys.argv[1])  # Choose this many elements
n = int(sys.argv[2])  # from 0, 1, ..., n-1.

# Initialize array perm = [0, 1, ..., n-1].
# perm = stdarray.create1D(n, 0)
perm = [0.0]*n
for i in range(n):
    perm[i] = i
print(perm)
    
# Create a random sample of size m in perm[0..m).
for i in range(m):
    r = random.randrange(i, n)
    print(r)

    
    # Exchange perm[i] and perm[r].
    temp = perm[r]
    perm[r] = perm[i]
    perm[i] = temp
print()

# Write the results.
for i in range(m):
    print(str(perm[i]) + ' ')
# stdio.writeln()

# stdio.writeln(m)

File: introCS/java/Sample.java
public class Sample
{
    public static void main(String[] args) 
    {
        int m = Integer.parseInt(args[0]);
        int n = Integer.parseInt(args[1]);
        int[] perm = new int[n];
        
        // Initialize perm[].
        for (int j = 0; j < n; j++)
            perm[j] = j;
            
        //System.out.println(perm);
        
        // Take sample.
        for (int i = 0; i < m; i++)
        {
            int r = i + (int) (Math.random() * (n-i));
            int t = perm[r];
            perm[r] = perm[i];
            perm[i] = t;
        }
        
        // Print sample.
        for (int i = 0; i < m; i++)
            System.out.print(perm[i] + " ");
        System.out.println();
    }
}